DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIImageResource.swift

```diff
@@ -0,0 +1,47 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+#if os(iOS)
+import UIKit
+
+internal struct UIImageResource {
+    internal let image: UIImage
+    internal let tintColor: UIColor?
+
+    internal init(image: UIImage, tintColor: UIColor?) {
+        self.image = image
+        self.tintColor = tintColor
+    }
+}
+
+extension UIImageResource: Resource {
+    func calculateIdentifier() -> String {
+        var identifier = image.dd.srIdentifier
+        if let tintColorIdentifier = tintColor?.srIdentifier {
+            identifier += tintColorIdentifier
+        }
+        return identifier
+    }
+
+    func calculateData() -> Data {
+        guard let tintColor = tintColor else {
+            return image.scaledDownToApproximateSize(SessionReplay.maxObjectSize)
+        }
+        if #available(iOS 13.0, *), image.isSymbolImage {
+            return image.withTintColor(tintColor)
+                .scaledDownToApproximateSize(SessionReplay.maxObjectSize)
+        } else {
+            return manuallyTintedImageData()
+        }
+    }
```

Comment:
suggestion/ add tests for identifier and data calculation

Reply:
Added some tests with soft assertions. Different simulators / iOS versions might produce different images / hashes.

It's more fully tested through snapshot tests that render this calculatedData() as images.